examples/chatgpt/gpt_actions_library/gpt_middleware_aws_function.ipynb

```diff
@@ -0,0 +1,421 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Value + Example Business Use Cases"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Value**: Users can now leverage ChatGPT's capabilities to connect to an AWS Function. This enables you to connect to any services in AWS and run code/applications on this. This can in a few ways:\n",
+    "\n",
+    "- Access 3rd party services such as AWS Redshift, AWS DynamoDB, AWS S3 and even more!\n",
+    "- Allows pre-processing text responses from an API (overcoming context limits, adding context or metadata as examples).\n",
+    "- Enables to return files instead of retrieving text from 3rd party APIs. This can be useful to surface CSV files for Data Analysis, or bring back an PDF file and ChatGPT will treat it like an upload. \n",
+    "\n",
+    "\n",
+    "**Example Use Cases**: \n",
+    "- A user needs to look up data in Redshift, but needs a middleware app between ChatGPT and Redshift to return files (data analysis data exactitude as well as large number of data)\n",
+    "- A user has built several steps in an AWS function, and needs to be able to kick off that process using ChatGPT."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Application information & prerequisites"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will leverage AWS Lambda services to create a middleware function. You can get familiar with this stack by visiting the following links: \n",
+    "\n",
+    "- Lambda Website: https://aws.amazon.com/lambda/\n",
+    "- Lambda Documentation: https://docs.aws.amazon.com/lambda/\n",
+    "- AWS SAM docs: https://docs.aws.amazon.com/serverless-application-model/"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prerequisites"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before you get started, make sure you have an AWS Console with access to create: Lambda Function, S3 Buckets, Application Stack, Cognito User Pool, Cognito User Pool App Clients, API Gateway, Lambda roles, CloudFormation stacks (this feels like a lot but creating those services is automated!)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create AWS Lambda Function"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To create an AWS Function you can use AWS SAM. An example of a SAM Template can be found [here](https://github.com/openai/redshift-middleware/blob/main/template.yaml) [0].\n",
```

Comment:
This is not a publicly accessible repo - should it be? Or should you remove the URL?

Reply:
good catch thanks!